src/users/users.service.ts

```diff
@@ -0,0 +1,101 @@
+import {
+    BadRequestException,
+    ForbiddenException,
+    Injectable,
+    NotFoundException,
+    UnauthorizedException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import * as bcrypt from 'bcrypt';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { UserEntity } from './user.entity';
+import { omit } from 'lodash';
+import { PostsService } from '../posts/posts.service';
+import { UserResponseDto } from './dto/user-response.dto';
+
+@Injectable()
+export class UsersService {
+    constructor(
+        @InjectRepository(UserEntity) private usersRepo: Repository<UserEntity>,
+        private postsService: PostsService,
+    ) {}
+
+    async create(data: CreateUserDto): Promise<UserEntity> {
+        const user = await this.usersRepo.create(data);
+        return this.usersRepo.save(user);
+    }
+
+    async findAll(email?: string): Promise<UserResponseDto[]> {
+        let users: UserEntity[];
+        if (email) {
+            users = await this.usersRepo.find({ where: { email: email.toLowerCase() } });
+        } else {
+            users = await this.usersRepo.find();
+        }
+        return users.map((user) => omit(user, 'password'));
+    }
+
+    async findByEmail(email: string) {
+        return this.usersRepo.findOne({ where: { email: email.toLowerCase() } });
+    }
+
+    async findOne(id: number): Promise<UserResponseDto> {
+        const user = await this.usersRepo.findOne({ where: { id } });
```

Comment:
Does typeorm have findById() function ? If so, let's use `findById(id)` it instead of `findOne({ where: { id } })`